packages/blue-sdk/src/position/PreLiquidatablePosition.ts

```diff
@@ -0,0 +1,330 @@
+import type { Address } from "viem";
+import { ORACLE_PRICE_SCALE } from "../constants";
+import {
+  CapacityLimitReason,
+  type Market,
+  MarketUtils,
+  type MaxBorrowOptions,
+  type MaxWithdrawCollateralOptions,
+} from "../market";
+import { MathLib, SharesMath } from "../math";
+import {
+  AccrualPosition,
+  type IAccrualPosition,
+  type IPosition,
+} from "./Position";
+
+export interface PreLiquidationParams {
+  preLltv: bigint;
+  preLCF1: bigint;
+  preLCF2: bigint;
+  preLIF1: bigint;
+  preLIF2: bigint;
+  preLiquidationOracle: Address;
+}
+
+export interface IPreLiquidatablePosition extends IPosition {
+  preLiquidationParams: PreLiquidationParams;
+  preLiquidation: Address;
+}
+
+export class PreLiquidatablePosition
+  extends AccrualPosition
+  implements IPreLiquidatablePosition
+{
+  /**
+   * The pre-liquidation parameters of the associated PreLiquidation contract.
+   */
+  public readonly preLiquidationParams: PreLiquidationParams;
+
+  /**
+   * The address of the PreLiquidation contract this position is associated to.
+   */
+  public readonly preLiquidation: Address;
+
+  constructor(
+    position: IAccrualPosition,
```

Comment:
```suggestion
    position: IPreLiquidatablePosition,
```

Reply:
The goal is to have only 2 parameters here, to avoid having 4 parameters and more if we ever extend this class again